rustls/src/common_state.rs

```diff
@@ -974,5 +979,40 @@ impl KxState {
     }
 }
 
+pub(crate) struct HandshakeFlight<'a, const TLS13: bool> {
+    pub(crate) transcript: &'a mut HandshakeHash,
+    body: Vec<u8>,
+}
+
+impl<'a, const TLS13: bool> HandshakeFlight<'a, TLS13> {
+    pub(crate) fn new(transcript: &'a mut HandshakeHash) -> Self {
+        Self {
+            transcript,
+            body: Vec::new(),
+        }
+    }
+
+    pub(crate) fn add(&mut self, hs: HandshakeMessagePayload<'_>) {
+        let start_len = self.body.len();
+        hs.encode(&mut self.body);
+        self.transcript
+            .add(&self.body[start_len..]);
+    }
+
+    pub(crate) fn finish(self, common: &mut CommonState) {
+        let m = Message {
+            version: if TLS13 {
+                ProtocolVersion::TLSv1_3
+            } else {
+                ProtocolVersion::TLSv1_2
+            },
+            payload: MessagePayload::handshake_flight(self.body),
+        };
+        common.send_msg(m, TLS13);
+    }
+}
+
+pub(crate) type HandshakeFlightTls12<'a> = HandshakeFlight<'a, false>;
```

Comment:
Is the TLS 1.2 use-case landed alongside the new types to avoid dead code warnings? I think this diff would be easier to digest if it were separated out like the TLS 1.3 case but I wager that might produce warnings and I don't feel very strongly.

Reply:
In general, not sure I'm convinced of the benefit of this type alias.

Reply:
Yes I would like to avoid the true/false thing leaking out of this file, as far as possible.

My second choice would be to stop this type being generic like this, and just have two different constructors (eg, `new_tls12()`, `new_tls13()` or similar)  and rely on everything being inlined into their uses.